lms/djangoapps/learner_home/test_utils.py

```diff
@@ -86,3 +88,19 @@ def create_test_enrollment(user, course_mode=CourseMode.AUDIT):
     test_enrollment.course_overview.end = random_date()
 
     return test_enrollment
+
+
+def expected_progress_url(course_id, course_home_mfe_progress_tab_is_active):
+    """
+    Returns expected course progress URL based on the status of the course_home_mfe_progress_tab_is_active.
+
+    :param course_id: The course key / ID for which the progress url is being requested.
+    :param course_home_mfe_progress_tab_is_active: A boolean matching the return value from the
+                                                   course_home_mfe_progress_tab_is_active function,
+                                                   implemented in lms.djangoapps.course_home_api.toggles.
+
+    :return: The course progress URL.
+    """
+    if course_home_mfe_progress_tab_is_active:
+        return f'{settings.LEARNING_MICROFRONTEND_URL}/course/{course_id}/progress'
```

Comment:
```suggestion
        return get_learning_mfe_home_url(course_id, url_fragment='progress')
```

Why not use the same underlying function, you're testing if the routing logic works so I don't see a reason to hardcode the URL here instead of using the function.

Reply:
Hi @feanil. I'm terribly sorry for answering this late. Had a couple of hard weeks with personal and health related issues. About your recommendations, you're right on both. Just something that comes to my mind is that after applying the suggestions, the `expected_progress_url` function in `test_utils.py` and `course_progress_url` function in `utils.py` are almost identical. Since there's no actual need to pass the state of `course_home_mfe_progress_tab_is_active` flag as an argument, my suggestion is to use the `course_progress_url` function in utils in the test as well and remove `expected_progress_url`. Is this ok in your opinion?

Reply:
Sorry to hear that @FatemeKhodayari hopefully you're feeling better. I think that's fine, at first I was concerned that it would just mean that we're comparing the value coming from `course_progress_url` with itself but I think it's fine.  We're making sure the value from that function matches what the CourseSerializer outputes at the end of serialization so I think it's a useful test.  Let's go ahead and just use the `course_progress_url` function during testing as well.

Reply:
> Sorry to hear that @FatemeKhodayari hopefully you're feeling better.

I'm all better now. Thanks @feanil.


> I think that's fine, at first I was concerned that it would just mean that we're comparing the value coming from `course_progress_url` with itself but I think it's fine. We're making sure the value from that function matches what the CourseSerializer outputes at the end of serialization so I think it's a useful test. Let's go ahead and just use the `course_progress_url` function during testing as well.

Yeah I thought the same. Done :)